### DISCUSSION TITLE
Pair scene and its script in File System section

### DISCUSSION THREAD

**Discussion by hoork** [godotengine/godot-proposals#4398] [Editor]
If scene was created and script attached to its root node, then display the script file as nested inside the scene file in file system.

Like Visual Studio do with XAML file and its code-behind.
![image_2022-04-08_15-10-16](https://user-images.githubusercontent.com/18242896/163252828-ef353b9f-1e94-4e50-b097-96db08eb2533.png)

Perhaps its should be applied if only the scene and the script are situated in the same directory and this particular script only attached to this one scene.

**Comment by Zireael07** [comment]
Scripts can be used in many scenes AND the official style guide recommends storing scripts separately, so I'm doubtful.

**Reply by hoork** [reply]
Can you give me link to this style guide? In [Project organization](https://docs.godotengine.org/en/stable/tutorials/best_practices/project_organization.html) page i found statement contrary to yours:

> Considering that, the most common approach is to group assets as close to scenes as possible; when a project grows, it makes it more maintainable.

Or did I misinterpret it?

**Reply by Mickeon** [reply]
Surely, but let's, for instance, create a "_Level.tscn_", containing common children ("Player", "Goal"...), with "_level.gd_" attached. Every level we are going to make will inherit from this base, and that includes the attached script, which doesn't necessarily have to change. In one (or more) of these levels you may even decide to create and attach the script "_level_opposite_day.gd_" which kills the Player when they reach the Goal, without needing to change the original Scene they inherit from.

So, it is true, it is recommended to keep them close, but Scenes and Scripts really can be separated from one another.

**Reply by Zireael07** [reply]
@Hoork: Notice it says "assets", not scripts

**Comment by Mickeon** [comment]
Scripts != Scenes, and same Scripts can really be used in wholy different Scenes, I'm not sure this would be a good idea, in general
